src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -28,24 +28,29 @@ public class HubConnectionContext
         private readonly ILogger _logger;
         private readonly CancellationTokenSource _connectionAbortedTokenSource = new CancellationTokenSource();
         private readonly TaskCompletionSource<object> _abortCompletedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
-        private readonly long _keepAliveDuration;
+        private readonly TimeSpan _keepAliveInterval;
+        private readonly TimeSpan _clientTimeoutInterval;
         private readonly SemaphoreSlim _writeLock = new SemaphoreSlim(1);
 
-        private long _lastSendTimestamp = Stopwatch.GetTimestamp();
+        private long _lastSendTimestamp = DateTime.UtcNow.Ticks;
```

Comment:
How come you're using UtcNow.Ticks instead of GetTimestamp?

Reply:
Kestrel actually uses `DateTime.UtcNow`, as does the client timeout logic. `DateTime.UtcNow` has basically the same performance, so we decided consistency was better here.